application/src/main/java/org/togetherjava/tjbot/commands/bytecode/BytecodeCommand.java

```diff
@@ -0,0 +1,269 @@
+
+    private final Pattern codeBlockExtractorPattern =
+            Pattern.compile("```(?:java)?\\s*([\\w\\W]+)```|``?([\\w\\W]+)``?");
+    private final Map<Long, List<Long>> userToBotMessages = new HashMap<>();
+
+    public BytecodeCommand() {
+        super(Commands.message("View bytecode"), CommandVisibility.GUILD);
+    }
+
+    // Fresh compile when a user uses the message context command on one of their messages
+    @Override
+    public void onMessageContext(@NotNull MessageContextInteractionEvent event) {
+        Message message = event.getTarget();
+        String content = message.getContentRaw();
+
+        event.reply("Compiling...")
+            .mentionRepliedUser(false)
+            .flatMap(InteractionHook::retrieveOriginal)
+            .queue(compReply -> compile(message, compReply, parseCommandFromMessage(content)));
+    }
+
+    // Delete our messages if the user deletes their request message
+    @Override
+    public void onMessageDeleted(@NotNull MessageDeleteEvent event) {
+        long mesageIdLong = event.getMessageIdLong();
+
+        if (userToBotMessages.containsKey(mesageIdLong)) {
+            deleteMyMessages(mesageIdLong, event.getTextChannel());
+        }
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        /* empty */
+    }
+
+    // Recompile when the user sends edits their request message
+    @Override
+    public void onMessageUpdated(@NotNull MessageUpdateEvent event) {
+        Message message = event.getMessage();
+        long messageIdLong = event.getMessageIdLong();
+
+        if (!userToBotMessages.containsKey(messageIdLong)) {
+            return;
+        }
+
+        TextChannel textChannel = message.getTextChannel();
+        List<Long> myMessages = userToBotMessages.get(messageIdLong);
+
+        if (myMessages.isEmpty()) {
+            message.reply(
+                    "An unknown error occurred (`userMessagesToBotMessages.get(messageIdLong).isEmpty() == true`)")
+                .queue();
+
+            return;
+        }
+
+        textChannel.retrieveMessageById(myMessages.get(0)).queue(myMessage -> {
+            textChannel.purgeMessagesById(myMessages.stream()
+                .skip(1) // skip our first message to edit it
+                .map(String::valueOf)
+                .toList());
+
+            myMessage.editMessage("Recompiling...").queue();
+
+            compile(message, myMessage, parseCommandFromMessage(message.getContentRaw()));
+        });
+    }
+
+    private void deleteMyMessages(@NotNull Long msgId, @NotNull TextChannel channel) {
+        if (!userToBotMessages.containsKey(msgId)) {
+            return;
+        }
+
+        channel
+            .purgeMessagesById(userToBotMessages.get(msgId).stream().map(String::valueOf).toList());
+
+        userToBotMessages.remove(msgId);
+    }
+
+    private void compile(@NotNull Message userMessage, @NotNull Message myMessage,
```

Comment:
who is `myXXX`? i remember that this confused me back then already. was it the bot? then please rename it to `botMessage` or similar. same with the occurences of `our`.

its confusing, as usually people assume that code is written from the perspective of the author, not the bot.